Remove EventSendingController::scalePageBy.

mattwoodrow opened the pull request:
#### 5b4882fba5fbf3315d93d45ee78f2d53e50e56d8
<pre>
Remove EventSendingController::scalePageBy.
<a href="https://bugs.webkit.org/show_bug.cgi?id=280166">https://bugs.webkit.org/show_bug.cgi?id=280166</a>
&lt;<a href="https://rdar.apple.com/136469621">rdar://136469621</a>&gt;

Reviewed by Kimmo Kinnunen.

This should be replaced by testRunner.setPageScaleFactor for site-isolation support.

* LayoutTests/compositing/fixed-with-fixed-layout.html:
* LayoutTests/compositing/layer-creation/zoomed-clip-intersection.html:
* LayoutTests/compositing/tiling/tile-cache-zoomed.html:
* LayoutTests/fast/css/sticky/sticky-top-zoomed-expected.html:
* LayoutTests/fast/css/sticky/sticky-top-zoomed.html:
* LayoutTests/fast/dom/zoom-scroll-page-test.html:
* LayoutTests/fast/events/ios/viewport-device-width-allows-double-tap-zoom-out.html:
* LayoutTests/fast/repaint/fixed-in-page-scale.html:
* LayoutTests/fast/repaint/fixed-right-bottom-in-page-scale.html:
* LayoutTests/fast/repaint/fixed-right-in-page-scale.html:
* LayoutTests/fast/repaint/zoomed-fixed-background.html:
* LayoutTests/fast/visual-viewport/zoomed-scroll-into-view-fixed.html:
* LayoutTests/fast/visual-viewport/zoomed-scroll-to-anchor-in-position-fixed.html:
* LayoutTests/platform/mac-wk2/plugins/mouse-events-scaled.html:
* LayoutTests/tiled-drawing/scrolling/fixed-background/fixed-body-background-zoomed.html:
* LayoutTests/tiled-drawing/scrolling/fixed/four-bars-zoomed.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-mandatory-mainframe-horizontal-with-keyboard-scaled.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-mandatory-mainframe-vertical-with-keyboard-scaled.html:
* LayoutTests/tiled-drawing/tile-size-slow-zoomed.html:
* LayoutTests/tiled-drawing/tiled-drawing-zoom-scrolled.html:
* LayoutTests/tiled-drawing/tiled-drawing-zoom.html:
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePage.cpp:
(WKBundlePageSetScaleAtOrigin): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePagePrivate.h:
* Tools/DumpRenderTree/mac/EventSendingController.mm:
(+[EventSendingController isSelectorExcludedFromWebScript:]):
(+[EventSendingController webScriptNameForSelector:]):
(-[EventSendingController scalePageBy:atX:andY:]): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/Bindings/EventSendingController.idl:
* Tools/WebKitTestRunner/InjectedBundle/EventSendingController.cpp:
(WTR::EventSendingController::scalePageBy): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/EventSendingController.h:
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundlePage.cpp:
(WTR::InjectedBundlePage::prepare):

Canonical link: <a href="https://commits.webkit.org/284532@main">https://commits.webkit.org/284532@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f870d9bac01d08b8dd5b5897214ea001b976df19

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/69526 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/48926 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/22199 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/73611 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/20684 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/71643 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/56727 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/20535 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/55279 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/13739 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/72592 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/44623 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/60023 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/35757 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/41289 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/17453 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/19061 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/63227 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/17799 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/75321 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/13508 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/17019 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/62947 "Found 1 new test failure: http/tests/inspector/network/har/har-page.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/13548 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/60106 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/62857 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/15484 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/10881 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/4503 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/44730 "Built successfully") | | | 
| | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/45804 "Build is in progress. Recent messages:OS: Sonoma (14.5), Xcode: 15.4; Running apply-patch; Checked out pull request; Compiled WebKit (warnings)") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/46999 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/45545 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

kkinnunen-apple commented on LayoutTests/platform/ios/TestExpectations:
```diff
@@ -7477,6 +7477,8 @@ imported/w3c/web-platform-tests/navigation-api/scroll-behavior/manual-scroll-rep
 [ Release arm64 ] fast/events/ios/activating-reset-button-should-not-scroll-page.html [ Skip ]
 [ Release arm64 ] fast/events/ios/activating-submit-button-should-not-scroll-page.html [ Skip ]
 
+webkit.org/b/107356 fast/css/sticky/sticky-top-zoomed.html [ ImageOnlyFailure ]
```
It was passing because the scale wasn't actually applied for the test nor the reference at the time of the pixel test?

mattwoodrow replied:
Yep! Already failing on every other platform, since the actual tested content appears to have been broken for a long time.

kkinnunen-apple commented on LayoutTests/platform/ios/compositing/tiling/tile-cache-zoomed-expected.txt:
```diff
@@ -4,18 +4,18 @@
   (coverage rect 0.00, 0.00 800.00 x 600.00)
   (intersects coverage rect 1)
-  (contentsScale 2.00)
+  (contentsScale 3.20)
```
It was passing because the scale was not actually applied at the time layer tree was snapshotted?

mattwoodrow replied:
Indeed. I didn't actually check the history here, but somehow we ended up with iOS-specific expectations that were clearly incorrect.

kkinnunen-apple reviewed (approved):
The expected changes could have used some explanation..

mattwoodrow committed:
Remove EventSendingController::scalePageBy.
https://bugs.webkit.org/show_bug.cgi?id=280166
<rdar://136469621>

Reviewed by Kimmo Kinnunen.

This should be replaced by testRunner.setPageScaleFactor for site-isolation support.

* LayoutTests/compositing/fixed-with-fixed-layout.html:
* LayoutTests/compositing/layer-creation/zoomed-clip-intersection.html:
* LayoutTests/compositing/tiling/tile-cache-zoomed.html:
* LayoutTests/fast/css/sticky/sticky-top-zoomed-expected.html:
* LayoutTests/fast/css/sticky/sticky-top-zoomed.html:
* LayoutTests/fast/dom/zoom-scroll-page-test.html:
* LayoutTests/fast/events/ios/viewport-device-width-allows-double-tap-zoom-out.html:
* LayoutTests/fast/repaint/fixed-in-page-scale.html:
* LayoutTests/fast/repaint/fixed-right-bottom-in-page-scale.html:
* LayoutTests/fast/repaint/fixed-right-in-page-scale.html:
* LayoutTests/fast/repaint/zoomed-fixed-background.html:
* LayoutTests/fast/visual-viewport/zoomed-scroll-into-view-fixed.html:
* LayoutTests/fast/visual-viewport/zoomed-scroll-to-anchor-in-position-fixed.html:
* LayoutTests/platform/mac-wk2/plugins/mouse-events-scaled.html:
* LayoutTests/tiled-drawing/scrolling/fixed-background/fixed-body-background-zoomed.html:
* LayoutTests/tiled-drawing/scrolling/fixed/four-bars-zoomed.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-mandatory-mainframe-horizontal-with-keyboard-scaled.html:
* LayoutTests/tiled-drawing/scrolling/scroll-snap/scroll-snap-mandatory-mainframe-vertical-with-keyboard-scaled.html:
* LayoutTests/tiled-drawing/tile-size-slow-zoomed.html:
* LayoutTests/tiled-drawing/tiled-drawing-zoom-scrolled.html:
* LayoutTests/tiled-drawing/tiled-drawing-zoom.html:
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePage.cpp:
(WKBundlePageSetScaleAtOrigin): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePagePrivate.h:
* Tools/DumpRenderTree/mac/EventSendingController.mm:
(+[EventSendingController isSelectorExcludedFromWebScript:]):
(+[EventSendingController webScriptNameForSelector:]):
(-[EventSendingController scalePageBy:atX:andY:]): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/Bindings/EventSendingController.idl:
* Tools/WebKitTestRunner/InjectedBundle/EventSendingController.cpp:
(WTR::EventSendingController::scalePageBy): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/EventSendingController.h:
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundlePage.cpp:
(WTR::InjectedBundlePage::prepare):

Canonical link: https://commits.webkit.org/284532@main
